File: backend/db/tasks/services/tasks.py
from db.tasks.models import Statistic, Task


class TaskService:
    def __init__(self, user_id):
        self.user_id = user_id

    async def get_statistic(self) -> Statistic:
        statistic, _ = await Statistic.get_or_create(user_id=self.user_id)
        return statistic

    async def create_tasks(self, text: str):
        statistic = await self.get_statistic()
        task = await Task.create(
            user_id=self.user_id,
            text=text
        )

        statistic.created += 1
        await statistic.save()

        return task

    async def delete_task(self, task: Task):
        statistic = await self.get_statistic()
        await task.delete()

        statistic.deleted += 1
        await statistic.save()

    async def check_task(self, task: Task):
        statistic = await self.get_statistic()
        statistic_diff = 1

        if task.finished:
            statistic_diff = -1

        task.finished = not task.finished
        statistic.finished += statistic_diff

        await task.save()
        await statistic.save()

        return task

    async def get_tasks(self):
        return Task.filter(user_id=self.user_id)

File: web/src/api/tasks.js
import axios from "./index";
import {returnData} from "./utils";

const baseUrl = "/tasks"

export function getStatistic() {
    return axios.get(`${baseUrl}/statistic/`)
        .then(returnData)
}

export function getTaskList() {
    return axios.get(`${baseUrl}/task/`)
        .then(returnData)
}

export function createTask(text) {
    return axios.post(`${baseUrl}/task/`, {text})
        .then(returnData)
}

export function checkTask(taskId) {
    return axios.put(`${baseUrl}/task/${taskId}`)
        .then(returnData)
}

export function deleteTask(taskId) {
    return axios.delete(`${baseUrl}/task/${taskId}`)
        .then(returnData)
}